lib/OpenQA/WebAPI/Controller/API/V1/Worker.pm

```diff
@@ -158,16 +160,30 @@ sub create {
     my $instance = $self->param('instance');
     my $caps     = {};
 
-    $caps->{cpu_modelname} = $self->param('cpu_modelname');
-    $caps->{cpu_arch}      = $self->param('cpu_arch');
-    $caps->{cpu_opmode}    = $self->param('cpu_opmode');
-    $caps->{mem_max}       = $self->param('mem_max');
-    $caps->{worker_class}  = $self->param('worker_class');
+    $caps->{cpu_modelname}                = $self->param('cpu_modelname');
+    $caps->{cpu_arch}                     = $self->param('cpu_arch');
+    $caps->{cpu_opmode}                   = $self->param('cpu_opmode');
+    $caps->{mem_max}                      = $self->param('mem_max');
+    $caps->{worker_class}                 = $self->param('worker_class');
+    $caps->{websocket_api_version}        = $self->param('websocket_api_version');
+    $caps->{isotovideo_interface_version} = $self->param('isotovideo_interface_version');
+
+    my $id;
+    try {
+        $id = $self->_register($self->db, $host, $instance, $caps);
+    }
+    catch {
+        if (/Incompatible/) {
+            $self->render(status => 426, json => {error => $_});
+        }
+        else {
+            die $_;
```

Comment:
just a question, there is a reason why here the webui process should die? (can't find it)

Reply:
@mudler 
this makes the server to throw a 500 error, which is what we want and it was also the previous behaviour.

Reply:
Then also 
```perl 
$self->render(json => {error => 'Failed ' . $error}, status => 500) and return if $error;
``` 
should work as well

Reply:
@mudler 
The tests i did while developing there was no difference between that and dying. I agree with you though. I'll change it to be more explicit.

Reply:
@mudler @foursixnine 

Changes done. Everything working as expected. Changes deployed in STG 2